File: app/src/main/java/com/sparknetwork/viewModel/UploadImagesViewModel.java
package com.sparknetwork.viewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.sparknetwork.model.ImageModel;
import com.sparknetwork.repository.UploadImagesRepository;

/**
 * Created by kamal on 8/2/18.
 */

public class UploadImagesViewModel extends ViewModel {

    private MutableLiveData<ImageModel> data;

    private UploadImagesRepository movieModel;

    public UploadImagesViewModel() {
        movieModel = new UploadImagesRepository();
    }

    public void init() {
        if (this.data != null) {
            return;
        }
        data = movieModel.getImages();
    }

    public MutableLiveData<ImageModel> getImages() {
        return this.data;
    }
}

File: app/src/main/java/com/sparknetwork/android/viewModel/UploadImagesViewModel.kt
package com.sparknetwork.viewModel

import android.app.Application
import android.net.Uri
import androidx.lifecycle.*
import com.sparknetwork.model.ImageModel
import com.sparknetwork.model.UploadProgress
import com.sparknetwork.repository.UploadImagesRepository

class UploadImagesViewModel(application: Application) : AndroidViewModel(application) {

    private val context = getApplication<Application>().applicationContext

    private lateinit var mRepository: UploadImagesRepository

    fun uploadImage(uri: Uri, mimeType: String?): MutableLiveData<UploadProgress> {
        mRepository = UploadImagesRepository(context)
        return mRepository.uploadImage(uri, mimeType)
    }

    fun getImages(): MutableLiveData<List<ImageModel>> {
        mRepository = UploadImagesRepository(context)
        return mRepository.getImageFiles()
    }
}